tests/functional/test_s3.py

```diff
@@ -462,19 +461,35 @@ def test_s3_copy_object_with_empty_response(self):
         self.assertEqual(response["ResponseMetadata"]["HTTPStatusCode"], 200)
         self.assertTrue("CopyObjectResult" in response)
 
-    def test_s3_copy_object_with_incomplete_response(self):
```

Comment:
Why did we remove the `test_s3_copy_object_with_incomplete_response` test?

Reply:
We no longer go through the parser first which will raise a `ResponseParserError` for an incomplete response. This operation now runs through the `check_for_200_error` handler first which will lead to retry a partial response instead of raising the error. 
Alternatively, we can check whether we retried and received a complete body, but the test above this one ([test_s3_copy_object_with_empty_response](https://github.com/boto/botocore/blob/develop/tests/functional/test_s3.py#L438C9-L463)) is already doing that with an empty body.